engine-ext/api/src/main/java/org/pentaho/di/engine/api/remote/StopMessage.java

```diff
@@ -25,50 +25,127 @@
-  //private CloseReason.CloseCode closeCode;
   private String reasonPhrase;
   private String requestUUID;
+  private Status result;
 
-  public StopMessage(
-    //CloseReason.CloseCode closeCode,
-    String reasonPhrase ) {
-    //this.closeCode = closeCode;
+  /**
+   * Constructor used by the clients when sending the stop request for the daemon server
+   *
+   * @param reasonPhrase reason for the Stop request
+   */
+  public StopMessage( String reasonPhrase ) {
     this.reasonPhrase = reasonPhrase;
+    this.result = Status.SUCCESS;
   }
 
-  public StopMessage(
-    String requestUUID,
-    String reasonPhrase ) {
+  /**
+   * Constructor used by the daemon server to send the stop request for the driver
+   *
+   * @param requestUUID  request/execution ID to stop
+   * @param reasonPhrase reason for the Stop request
+   */
+  public StopMessage( String requestUUID, String reasonPhrase ) {
     this.requestUUID = requestUUID;
     this.reasonPhrase = reasonPhrase;
+    this.result = Status.SUCCESS;
   }
 
-  public String getReasonPhrase() {
-    return reasonPhrase;
+  /**
+   * Constructor used by driver to send back to daemon server the sessionKilled of the stop operation
+   *
+   * @param reasonPhrase returns back the reason presented by daemon server for the stop request
+   * @param result       stop operation result: SUCCESS, FAILED, SESSION_KILLED;
+   */
+  public StopMessage( String reasonPhrase, Status result ) {
+    this.reasonPhrase = reasonPhrase;
+    this.result = result;
+  }
+
+  /**
+   * Constructor used by driver to send back to daemon server the sessionKilled of the stop operation
+   *
+   * @param requestUUID  request/execution ID to stop or stopped
+   * @param reasonPhrase returns back the reason presented by daemon server for the stop request
+   * @param result       stop operation result: SUCCESS, FAILED, SESSION_KILLED;
+   */
+  public StopMessage( String requestUUID, String reasonPhrase, Status result ) {
+    this.requestUUID = requestUUID;
+    this.reasonPhrase = reasonPhrase;
+    this.result = result;
   }
 
+  /**
+   * Returns the request unique identifier
+   *
+   * @return request unique identifier
+   */
   public String getRequestUUID() {
     return requestUUID;
   }
 
+  /**
+   * Sets the stop reason phrase
+   *
+   * @param reasonPhrase stop reason phrase
+   */
   public void setReasonPhrase( String reasonPhrase ) {
     this.reasonPhrase = reasonPhrase;
   }
 
-  //    public CloseReason toCloseReason() {
-  //        return new CloseReason(closeCode, reasonPhrase);
-  //    }
+  /**
+   * Returns the stop reason.
+   *
+   * @return stop reason;
+   */
+  public String getReasonPhrase() {
+    return reasonPhrase;
+  }
+
+  /**
+   * True if the execution was stop with success, false otherwise.
+   *
+   * @return true if failed to stop the execution, false otherwise;
+   */
+  public boolean operationSuccessful() {
```

Comment:
small nitpick thing:  consider changing to "isOperationSuccessful()".  Follow pattern of boolean getters prefixed with "is"